packages/cockroach/extra/cockroachdb-change-config.py

```diff
@@ -20,64 +20,70 @@
+
+    In order to adjust this to the maximum replication factor possible for a
+    given DC/OS cluster and to provide maximum fault-tolerance we set the
+    cluster-wide configuration setting `num_replicas` for any database entity
+    to the given `num_replicas` count which equals the number of DC/OS master
+    nodes.
     Relevant JIRA ticket: https://jira.mesosphere.com/browse/DCOS-20352
     """
-
-    def _set_replicas_for_zone(zone: str) -> None:
-        command = [
-            '/opt/mesosphere/active/cockroach/bin/cockroach',
-            'zone',
-            'set',
-            zone,
-            '--insecure',
-            '--host={}'.format(my_internal_ip),
-            '-f',
-            '-'
-            ]
-        config_text = 'num_replicas: %s' % (num_replicas, )
-        log.info('Set `%s` via command `%s`', config_text, ' '.join(command))
-        subprocess.run(command, input=config_text.encode('ascii'))
+    zone_config = 'num_replicas = {}'.format(num_replicas)
+
+    # If more entities that must have their `num_replicas` setting adjusted are
+    # added to CockroachDB, a DC/OS check will fail on all clusters (DC/OS OSS
+    # 1.13+, DC/OS Enterprise 1.12+).
+    # describing that there are "underreplicated ranges".
+    # The DC/OS check is named "cockroachdb_replication".
+    #
+    # To resolve this, add more items to the following list.
+    # To display CockroachDB entities that must have their `num_replicas` setting
+    # adjusted, issue the following SQL command: `SHOW ALL ZONE CONFIGURATIONS;`
+    #
+    # One option is to parse the output of the above SQL command.
+    # However, there is a plan for CockroachDB that all replication will be
+    # derived from ``.default``.
+    # See https://forum.cockroachlabs.com/t/change-replication-factor/2052/3.
+    # Should this happen, we can replace the following loop initialisation with:
+    # zone = '.default'
+    # db_entity = 'RANGE default'
+    for zone, db_entity in [
+        ('.default', 'RANGE default'),
+        ('system', 'DATABASE system'),
+        ('system.jobs', 'TABLE system.public.jobs'),
+        ('.meta', 'RANGE meta'),
+        ('.system', 'RANGE system'),
+        ('.liveness', 'RANGE liveness'),
+    ]:
+        sql_command = (
+            'ALTER {db_entity} CONFIGURE ZONE USING {zone_config};'
+        ).format(
+            db_entity=db_entity,
+            zone_config=zone_config,
+        )
+        command = (
+            '/opt/mesosphere/active/cockroach/bin/cockroach '
+            'sql -e "{sql_command}" --insecure --host={host}'
+        ).format(
+            sql_command=sql_command,
+            host=my_internal_ip,
+        )
+        message = (
+            'Set {zone_config} for {zone} via command {command}'
+        ).format(
+            zone_config=zone_config,
+            zone=zone,
+            command=command,
+        )
+        log.info(message)
+        subprocess.run(command, shell=True)
```

Comment:
Let's keep in mind that if we are going to do a lot of work on this we should use Python rather than Bash - https://www.cockroachlabs.com/docs/stable/build-a-python-app-with-cockroachdb.html#basic-statements.